db/migrate/20170519153350_add_showback_models.rb

```diff
@@ -0,0 +1,73 @@
+class AddShowbackModels < ActiveRecord::Migration[5.0]
+  def up
+    create_table :showback_usage_types do |t|
+      t.string     :category
+      t.string     :description
+      t.string     :measure
+      t.text       :dimensions
+      t.timestamps
+    end
+
+    create_table :showback_events do |t|
+      t.jsonb      :data
+      t.timestamp  :start_time
+      t.timestamp  :end_time
+      t.belongs_to :resource, :type => :bigint, :polymorphic => true, :index => true
+      t.jsonb      :context
+      t.timestamps
+    end
+
+    create_table :showback_price_plans do |t|
+      t.string     :name
+      t.belongs_to :resource, :type => :bigint, :polymorphic => true
+      t.string     :description
+      t.timestamps
+    end
+
+    create_table :showback_rates do |t|
+      t.bigint     :fixed_rate_cents # Columns needed by gem money
```

Comment:
Int has a maximum value of  2_147_483_647, however, Money stores cents in that column, not ints, so the maximum is 21_474_836,47 $, it looks high, but it is not enough as we have some customers alreasy paying that amount on their monthly invoices (and everything yearly will be higher).
If you think that there are currencies that use 3 decimal digits, that gets reduced to 2M, and that is way too low.
For CCSP use cases, where you can have  multiple customers that you are serving, hardware management, etc. it gets worse.